tests/chainer_tests/functions_tests/array_tests/test_cast.py

```diff
@@ -57,4 +57,21 @@ def test_backward_cpu(self):
         self.check_backward(self.x, self.g)
 
 
+class TestNoCast(unittest.TestCase):
+
+    def setUp(self):
+        self.dtype = numpy.float32
+        self.x = numpy.empty(1, self.dtype)
+
+    def check_forward_no_cast(self, x_data):
+        y = functions.cast(x_data, self.dtype)
+        self.assertIs(y, x_data)
```

Comment:
That makes sense. I really want to make `ndarray` and `Variable` transparently exchanged, though... maybe it is impossible to do as a future extension of Chainer without big compatibility break... OK, I will fix this PR to always wrap the result by Variable.